File: src/main/java/kr/masker/core/util/RegexExpression.java
package kr.masker.core.util;

public enum RegexExpression {
    EMAIL("(?<=.{3}).(?<!@.?)"),
    PHONE("(?<=.{3}).(?=.{4})"),
    NAME_LESS_THAN_THREE("(.)$"),
    NAME_GREATER_THAN_OR_EQUAL_THREE("(?<=.).(?=.)"),
    ALL(".");
    private final String value;

    RegexExpression(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }
}

File: core/src/main/kotlin/constant/RegexExpression.kt
object RegexExpression {
    const val EMAIL: String = "(?<=.{3}).(?<!@.?)"
    const val PHONE: String = "(?<=.{3}).(?=.{4})"
    const val NAME_LESS_THAN_THREE: String = "(.)\$"
    const val NAME_NOT_LESS_THAN_THREE: String = "(?<=.).(?=.)"
    const val SELECT_ALL: String = "."
}
